### ISSUE TITLE
No link-local IPv6 addresses used for call-me-maybe

### ISSUE THREAD

**Issue by flub** [n0-computer/iroh#2491]
The `DiscoveredEndpoints` struct (needs rename, no "endpoint") is responsible for collecting the direct addresses which are sent in a call-me-maybe message.  But they do not include link-local IPv6 addresses.  This means we will not discover the most direct route on an IPv6 local network.  There really is not much downside to including these addresses.

**Comment by matheus23** [comment]
If iroh gets these addresses, since #3770 it will now actually make use of them. That said, for iroh to actually *obtain* these messages, it might be necessary to merge this PR in net-tools: https://github.com/n0-computer/net-tools/pull/50

**Comment by flub** [comment]
Yes, that net-tools change would need to be added. I also worry that we would need to start doing some form of pairing in holepunching: when we try a link-local dst we should only do so if we have a matching link-local src that is on the same subnet/link. And I don't think we have that logic yet.
